src/components/ChangeLocation/component/Location/index.js

```diff
@@ -1,71 +1,197 @@
+  useEffect(() => {
+    if (location) {
+      setPaths([...paths, location]);
+    }
+
+    if (chosenPath) {
+      setChosenCategoryPathID(chosenPath.id);
+      setPaths([...paths, chosenPath]);
+      setChosenPath(null);
+    }
+  }, [location, chosenPath]);
 
   return (
     <Form>
       <Form.Group className="mb-3" controlId="formBasicEmail">
         <Form.Label className={style.locationLabel}>Location</Form.Label>
-        <Form.Control readonly="readonly" value="Programming" />
+        <Form.Control readOnly="readonly" value={pathDisplay} />
       </Form.Group>
 
       <ListGroup>
-        {categories &&
+        {!chosenCategoryPathID && categories && type === 'Category' ? (
+          <ListGroup.Item
+            className={style.categoryListItem}
+            onMouseEnter={() => {
+              setHoveredItem(-1);
+            }}
+            onMouseLeave={() => {
+              setHoveredItem(null);
+            }}
+          >
+            <div>Root</div>
+            <BsArrowRightCircle
+              className={hoveredItem === -1 ? style.arrowRightIcon : 'd-none'}
+              onClick={() => {
+                setLocation(null);
+                setPathDisplay('Root');
+              }}
+            />
+          </ListGroup.Item>
+        ) : (
+          ''
+        )}
+        {categories ? (
           categories.map((category, idx) => {
             return (
               <ListGroup.Item
                 key={idx}
                 className={style.categoryListItem}
-                onClick={() => setLocation([...location, category.name])}
                 onMouseEnter={() => {
                   setHoveredItem(idx);
                 }}
                 onMouseLeave={() => {
                   setHoveredItem(null);
                 }}
               >
-                {category.name}
+                <div
+                  onClick={() => {
+                    if (category.subcategories_count > 0) {
+                      setPaths(
+                        paths.filter((path) => path.id !== location?.id)
+                      );
+                      setLocation(null);
+                      setChosenPath(category);
+                    } else {
+                      toast(
+                        'Message',
+                        'This category does not have a subcategory.'
+                      );
+                    }
+                  }}
+                >
+                  {category.name}
+                </div>
                 <BsArrowRightCircle
                   className={
                     hoveredItem === idx ? style.arrowRightIcon : 'd-none'
                   }
+                  onClick={() => {
+                    if (location?.name !== category.name) {
+                      setPaths(
+                        paths.filter((path) => path.id !== location?.id)
+                      );
+                      setLocation(category);
+                    }
```

Comment:
maybe you can refactor this with the same function from lines 143 to 156 since they are using some similar codes. but if is too complicated, leave as is